pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -45,16 +48,29 @@ var (
 )
 
 type splitAndCacheMiddlewareMetrics struct {
-	splitQueriesCount prometheus.Counter
+	splitQueriesCount            prometheus.Counter
+	queryResultCacheSkippedCount *prometheus.CounterVec
 }
 
 func newSplitAndCacheMiddlewareMetrics(reg prometheus.Registerer) *splitAndCacheMiddlewareMetrics {
-	return &splitAndCacheMiddlewareMetrics{
+	m := &splitAndCacheMiddlewareMetrics{
 		splitQueriesCount: promauto.With(reg).NewCounter(prometheus.CounterOpts{
 			Name: "cortex_frontend_split_queries_total",
-			Help: "Total number of underlying query requests after the split by interval is applied",
+			Help: "Total number of underlying query requests after the split by interval is applied.",
 		}),
+		queryResultCacheSkippedCount: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_frontend_query_result_cache_skipped_total",
+			Help: "Total number of times a query was not cacheable because of a reason.",
```

Comment:
```suggestion
			Help: "Total number of times a query was not cacheable because of a reason. This metric is tracked for each partial query when time-splitting is enabled.",
```